integration/test/basic/basic_test.go

```diff
@@ -14,35 +14,53 @@ import (
 	"github.com/giantswarm/chart-operator/integration/templates"
 )
 
-func TestChartInstalled(t *testing.T) {
-	err := f.InstallResource("chart-operator-resource", templates.ChartOperatorResourceValues, ":stable")
+func TestChartLifecycle(t *testing.T) {
+	const testRelease = "tb-release"
+	const cr = "chart-operator-resource"
+
+	log.Printf("creating %q", cr)
+	err := f.InstallResource(cr, templates.ChartOperatorResourceValues, ":stable")
+	if err != nil {
+		t.Fatalf("could not install %q %v", cr, err)
+	}
+
+	err = waitForReleaseStatus(helmClient, testRelease, "DEPLOYED")
+	if err != nil {
+		t.Fatalf("could not get release status of %q %v", testRelease, err)
+	}
+	log.Printf("%q succesfully deployed", testRelease)
+
+	log.Printf("deleting %q", cr)
+	err = helmClient.DeleteRelease(cr)
 	if err != nil {
-		t.Fatalf("could not install chart-operator-resource-chart %v", err)
+		t.Fatalf("could not delete %q %v", cr, err)
 	}
 
-	var rc *helmclient.ReleaseContent
+	err = waitForReleaseStatus(helmClient, testRelease, "DELETED")
+	if !helmclient.IsReleaseNotFound(err) {
+		t.Fatalf("%q not succesfully deleted %v", testRelease, err)
+	}
+	log.Printf("%q succesfully deleted", testRelease)
+}
+
+func waitForReleaseStatus(helmClient *helmclient.Client, release string, status string) error {
 	operation := func() error {
-		rc, err = helmClient.GetReleaseContent("tb-release")
+		rc, err := helmClient.GetReleaseContent(release)
 		if err != nil {
 			return microerror.Maskf(err, "could not retrieve release content")
 		}
-		if rc.Status == "PENDING_INSTALL" {
-			return microerror.Newf("release still not installed")
+		if rc.Status != status {
+			return microerror.Newf("waiting for %q, current %q", status, rc.Status)
 		}
 		return nil
 	}
 
 	notify := func(err error, t time.Duration) {
-		log.Printf("waiting for release %s: %v", t, err)
+		log.Printf("getting release status %s: %v", t, err)
 	}
 
-	err = backoff.RetryNotify(operation, backoff.NewExponentialBackOff(), notify)
-	if err != nil {
-		t.Fatal("expected nil found", err)
-	}
+	eb := backoff.NewExponentialBackOff()
+	eb.MaxElapsedTime = time.Minute
```

Comment:
Don't wait for the default `15` minutes, since we need to match the error, and 1 minute has been enough so far. We can iteratate on that.